File: cpp/945.minimum-increment-to-make-array-unique.cpp
#include <algorithm>
#include <vector>

using std::sort;
using std::vector;

class Solution {
public:
  int minIncrementForUnique(vector<int> &nums) {
    int prev = -1;
    int ans = 0;
    sort(nums.begin(), nums.end());

    for (int i = 0; i < nums.size(); ++i) {
      if (prev < nums[i]) {
        prev = nums[i];
        continue;
      }

      ans += prev - nums[i] + 1;
      prev = prev + 1;
    }

    return ans;
  }
};

File: python/945.minimum-increment-to-make-array-unique.py
from typing import List


class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        res = 0
        prev = -1
        nums.sort()

        for i in range(len(nums)):
            if prev < nums[i]:
                prev = nums[i]
                continue

            diff = prev - nums[i] + 1
            res += diff
            prev = nums[i] + diff
        return res

